exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+	params.Set("toEmail", arg.ToEmail)
+	params.Set("asset", arg.Asset)
+	params.Set("amount", fmt.Sprintf("%f", arg.Amount))
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	return &response, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary, http.MethodPost, subaccountTransfer, params, spotDefaultRate, &response)
+}
+
+// GetSubaccountAssets to fetch sub-account assets.
+func (bi *Binanceus) GetSubaccountAssets(ctx context.Context, email string) (*SubAccountAssets, error) {
+	var resp SubAccountAssets
+	if !common.MatchesEmailPattern(email) {
+		return nil, errNotValidEmailAddress
+	}
+	params := url.Values{}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", fmt.Sprintf("%d", timestamp))
+	params.Set("email", email)
+	//
+	return &resp, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary, http.MethodGet,
+		subaccountAssets, params,
+		spotDefaultRate,
+		&resp)
+}
+
+// GetMasterAccountTotalUSDValue this endpoint to get the total value of assets in the master account in USD.
+func (bi *Binanceus) GetMasterAccountTotalUSDValue(ctx context.Context, email string, page, size int) (*SpotUSDMasterAccounts, error) {
+	var response SpotUSDMasterAccounts
+	params := url.Values{}
+	if email != "" {
+		params.Set("email", email)
+	}
+	if page > 0 {
+		params.Set("page", strconv.Itoa(page))
+	}
+	if size > 0 {
+		params.Set("size", strconv.Itoa(size))
+	}
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	return &response, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary,
+		http.MethodGet, masterAccounts, params,
+		spotDefaultRate, &response)
+}
+
+// GetSubaccountStatusList this endpoint retrives a status list of sub-accounts.
+func (bi *Binanceus) GetSubaccountStatusList(ctx context.Context, email string) ([]SubAccountStatus, error) {
+	params := url.Values{}
+	if !common.MatchesEmailPattern(email) {
+		return nil, errMissingSubAccountEmail
+	}
+	params.Set("email", email)
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	var response []SubAccountStatus
+	return response, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary,
+		http.MethodGet, subAccountStatusList, params,
+		spotDefaultRate, &response)
+}
+
+// Trade Order Endpoints
+
+// GetOrderRateLimits get the current trade order count rate limits for all time intervals.
+// INPUTS: recvWindow <= 60000
+func (bi *Binanceus) GetOrderRateLimits(ctx context.Context, recvWindow uint) ([]OrderRateLimit, error) {
+	params := url.Values{}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	if recvWindow > 1000 && recvWindow < 60000 {
+		params.Set("recvWindow", strconv.Itoa(int(recvWindow)))
+	} else {
+		params.Set("recvWindow", strconv.Itoa(30000))
+	}
+	var resp []OrderRateLimit
+	return resp, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary, http.MethodGet, orderRateLimit, params, spotDefaultRate, &resp)
```

Comment:
The Get order rate limits endpoint has a request weight of 20, not 1. See https://docs.binance.us/#get-order-rate-limits